# Урок 4. Домашнее задание

## Непрерывные случайные величины. Функция распределения и плотность распределения вероятностей. Равномерное и нормальное распределение. Центральная предельная теорема

In [16]:
import numpy as np
import pandas as pd
import scipy.stats as stats
from matplotlib import pyplot as plt

### Задача 4.1

О непрерывной равномерно распределенной случайной величине `B` известно, что ее дисперсия равна 0.2. Можно ли найти правую границу величины `B` и ее среднее значение зная, что левая граница равна 0.5? Если да, найдите ее.

#### Решение:

$D(X) = 0.2$, Величина `B` находится на отрезке [a,b] =  [0.5, ?], т.е. $a = 0.5$, $b = ?$.

Математическое ожидание и дисперсия равномерного распределения равны:

$M(X) = \dfrac{a + b}{2}, \text{ } D(X) = \dfrac{(b - a)^2}{12}.$

Зная дисперсию $D(X)$ и начало отрезка $a$ можно выразить конец отрезка $b$, а далее и математическое ожидание $M(X)$ (читай среднее):

$b = \sqrt {12 \cdot D(X)} + a $

In [7]:
a = 0.5
d = 0.2

b = np.math.sqrt(12 * d) + a
b

2.049193338482967

In [9]:
m = (a + b) / 2
m

1.2745966692414834

#### Ответ:

Правая граница $b \approx 2.049$

Среднее значение (математическое ожидание) $M(X) \approx 1.275$



### Задача 4.2

На сколько сигм (средних квадратических отклонений) отклоняется рост человека, равный 190 см, от математического ожидания роста в популяции, в которой `M(X) = 178` и `D(X) = 25`?

#### Решение:

Рост человека имеет нормальное распределение непрерывной случайной величины. Воспользуемся соответствующими формулами для поиска сигмы $\sigma$:

$M(X) = a, \text{ } D(X) = \sigma^2.$

$\sigma = \sqrt {D(X)} = 5$

In [11]:
(190 - 178) / 5

2.4

#### Ответ:

Рост человека, равный 190 см отклоняется от математического ожидания на 2.4 сигм.

### Задача 4.3

Рост взрослого населения города `X` имеет нормальное распределение, причем, средний рост равен 174 см, а среднее квадратическое отклонение равно 8 см. Какова вероятность того, что случайным образом выбранный взрослый человек имеет рост:
1. больше 182 см?
2. больше 190 см?
3. от 166 см до 190 см?
4. от 166 см до 182 см?
5. от 158 см до 190 см?
6. не выше 150 см или не ниже 190 см?
7. не выше 150 см или не ниже 198 см?
8. ниже 166 см?

#### Решение:

Рост человека имеет нормальное распределение непрерывной случайной величины. Соответственно:

$a = 174$, 

$\sigma = 8.$

__1. больше 182 см?__

Интервал от $a - \sigma$ до $a + \sigma$ содержит около 68% вероятностной массы (__правило "одной" сигмы__). Т.е. интервал [174 - 8, 174 + 8] = [166, 182] содержит 68% вероятностной массы. На хвосты остается 32% вероятности. Вероятность того, что рост случайным образом выбранного человека будет больше 182 см (правый хвост) $\approx  \dfrac {0.32} {2} = 0.16$.

Проверим с помощью встроенного метода модуля `scipy.stats`:

In [26]:
stats.norm.cdf(x=182, loc=174, scale=8)

0.8413447460685429

Вероятность что рост будет до 182 см равна $\approx 0.84$, а значит вероятность, что рост будет больше 182 см = 1 - 0.84 = 0.16

__2. больше 190 см?__

Интервал от $a - 2\sigma$ до $a + 2\sigma$ — около 95% массы (__правило двух сигм__) = [158, 190]. Вероятность того, что рост человека будет больше 190 см $ \approx \dfrac {1 -  0.95} {2} = 0.025$

Проверим с помощью встроенного метода модуля `scipy.stats`:

In [25]:
1 - stats.norm.cdf(x=190, loc=174, scale=8)

0.02275013194817921

__3. от 166 см до 190 см?__

Интервал [166,182] попадает в правило одной сигмы, интервал [158, 190] в правило двух сигм. Значит искомый нами интервал это интервал правила одной сигмы и половина разности вероятностей двух интервалов (двух сигм минус одной сигмы):

In [32]:
68 + (95 - 68) / 2

81.5

Проверим с помощью встроенного метода модуля `scipy.stats`:

In [30]:
stats.norm.cdf(x=190, loc=174, scale=8) - stats.norm.cdf(x=166, loc=174, scale=8)

0.8185946141203637

__4. от 166 см до 182 см?__

По правилу одной сигмы вероятность равна $\approx 0.68$

Проверим с помощью встроенного метода модуля `scipy.stats`:

In [35]:
stats.norm.cdf(x=182, loc=174, scale=8) - stats.norm.cdf(x=166, loc=174, scale=8)

0.6826894921370859

__5. от 158 см до 190 см?__

По правилу двух сигм вероятность равна $\approx 0.95$

Проверим с помощью встроенного метода модуля `scipy.stats`:

In [34]:
stats.norm.cdf(x=190, loc=174, scale=8) - stats.norm.cdf(x=158, loc=174, scale=8)

0.9544997361036416

__6. не выше 150 см или не ниже 190 см?__

Посчитаем хвосты интервалов трех [150, 198] и двух сигм [158, 190], поделим их пополам и сплюсуем:

In [36]:
(1 - 0.997) / 2 + (1 - 0.95) / 2

0.026500000000000024

Проверим с помощью встроенного метода модуля `scipy.stats`:

In [41]:
stats.norm.cdf(x=150, loc=174, scale=8) + (1 - stats.norm.cdf(x=190, loc=174, scale=8))

0.0241000299798093

__7. не выше 150 см или не ниже 198 см?__

Это хвосты интервала трех сигм. Соответственно:

In [42]:
1 - 0.997

0.0030000000000000027

Проверим с помощью встроенного метода модуля `scipy.stats`:

In [43]:
stats.norm.cdf(x=150, loc=174, scale=8) + (1 - stats.norm.cdf(x=198, loc=174, scale=8))

0.0026997960632601965

__8. ниже 166 см?__

Эта вероятность равна половине хвоста правила одной сигмы:

In [44]:
(1 - 0.68) / 2

0.15999999999999998

Проверим с помощью встроенного метода модуля `scipy.stats`:

In [45]:
stats.norm.cdf(x=166, loc=174, scale=8)

0.15865525393145707